dspace-api/src/main/java/org/dspace/storage/bitstore/AzureBlobStorageBitStoreService.java

```diff
@@ -0,0 +1,395 @@
+			bitstream.setChecksumAlgorithm(this.algorithm);
+
+			scratchFile.delete();
+
+		} catch (Throwable t) {
+			log.error("put(" + bitstream.getInternalId() + ", is)", t);
+
+			StringWriter stringWriter = new StringWriter();
+			PrintWriter printWriter = new PrintWriter(stringWriter);
+			t.printStackTrace(printWriter);
+			if (t instanceof StorageException) {
+				if (((StorageException) t).getExtendedErrorInformation() != null) {
+					log.error(String.format("\nError: %s",
+							((StorageException) t).getExtendedErrorInformation().getErrorMessage()), t);
+				}
+			}
+			log.error(String.format("Exception details:\n%s", stringWriter.toString()), t);
+			throw new IOException(t);
+		} finally {
+			if (scratchFile.exists()) {
+				scratchFile.delete();
+			}
+		}
+	}
+
+	/**
+	 * Obtain technical metadata about an asset in the asset store.
+	 *
+	 * Checksum used is 128-bit MD5 digest.
+	 *
+	 * @param bitstream
+	 *            The asset to describe
+	 * @param attrs
+	 *            A Map whose keys consist of desired metadata fields
+	 *
+	 * @exception java.io.IOException
+	 *                If a problem occurs while obtaining metadata
+	 * @return attrs A Map with key/value pairs of desired metadata If file not
+	 *         found, then return null
+	 */
+	@Override
+	public Map about(Bitstream bitstream, Map attrs) throws IOException {
+		String key = getFullKey(bitstream.getInternalId());
+		try {
+			CloudBlob blob = this.blobContainer.getBlockBlobReference(key);
+			blob.downloadAttributes();
+
+			if (blob != null) {
+				BlobProperties blobProps = blob.getProperties();
+
+				if (blobProps != null) {
+					Date lastModified = blobProps.getLastModified();
+					String lenght = String.valueOf(blobProps.getLength());
+					String checksum = blobProps.getEtag();
+					String lastModifiedTimeString = null;
+
+					if (lastModified != null) {
+						lastModifiedTimeString = String.valueOf(lastModified.getTime());
+					}
+
+					String newLine = System.getProperty("file.separator");
+
+					log.debug("blobProperties:" + newLine + "lenght: " + lenght + newLine + "checksum: " + checksum
+							+ newLine + "last modified: " + lastModifiedTimeString);
+
+					log.debug("Attributes raw: " + attrs);
+
+					if (attrs != null) {
+
+						log.debug("Attributes as String: " + attrs.toString());
+
+						if (lenght != null && attrs.containsKey("size_bytes")) {
+							attrs.put("size_bytes", lenght);
+						}
+
+						if (checksum != null && attrs.containsKey("checksum")) {
+							attrs.put("checksum", checksum);
+							attrs.put("checksum_algorithm", this.algorithm);
+						}
+
+						if (lastModifiedTimeString != null && attrs.containsKey("modified")) {
+							attrs.put("modified", lastModifiedTimeString);
+						}
+
+						return attrs;
+					} else {
+						// not sure if this would ever happen
+						return null;
+					}
+				}
+			}
+
+		} catch (Exception e) {
+			log.error("about(" + key + ", attrs)", e);
+			throw new IOException(e);
```

Comment:
Similar to all the log.error comments, is there more information we should add to this newly thrown IOException?